### DISCUSSION TITLE
Invalidation service

### DISCUSSION THREAD

**Discussion by SebastianStehle** [ZarehD/AspNetStatic#10] [Ideas]
Hi,

is there an invalidation service? e.g. I have webhooks that can be used to refresh the content and I would like to integrate every page.

**Comment by ZarehD** [comment]
Hi, @SebastianStehle.

Please correct me if I misunderstood, but what you asking about is a way to manually trigger generation for an individual/specific page. That's an interesting (and useful) feature, but there's currently no such facility for that. For now, I would recommend you use the periodic re-generation facility. In the meanwhile, I will look into adding the feature.

**Comment by SebastianStehle** [comment]
Yes, exactly this is what I am looking for.

**Reply by ZarehD** [reply]
Hi, @SebastianStehle.

I implemented the feature you described, a new `IHost` extension method called `GenerateStaticPage`. Install v0.18.0 to get the code. Enjoy!

**Reply by SebastianStehle** [reply]
Thanks, but I think this is to complicated. I think it would make more sense to implement something like surrogate keys: https://docs.fastly.com/en/guides/working-with-surrogate-keys

A page could return them as headers or something like that (extension method would be helpful) and you create a file as reverse index: surrogate-key => urls[]
